src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java

```diff
@@ -324,7 +324,16 @@ protected ResponseEntity searchUsersByOrganisation(String organisationIdentifier
         return responseEntity;
     }
 
-    protected ResponseEntity<ModifyUserRolesResponse> modifyRolesForUserOfOrganisation(UserProfileUpdatedData userProfileUpdatedData, String userId, Optional<String> origin) {
+    protected ResponseEntity<ModifyUserRolesResponse> modifyRolesForUserOfOrganisation(UserProfileUpdatedData userProfileUpdatedData, String userId, Optional<String> origin, String orgId) {
+
+        organisationCreationRequestValidator.validateOrganisationIdentifier(orgId);
+        Organisation organisation = organisationService.getOrganisationByOrgIdentifier(orgId);
+
+        if (null == organisation) {
+            String errorMessage = "Unable to modify User Roles as no Organisation was found with the given organisationIdentifier: " + orgId;
+            log.error(errorMessage);
+            throw new InvalidRequest(errorMessage);
```

Comment:
should not this throw 404 if the organisation returned from db is null?

Reply:
Addressed in this commit: https://github.com/hmcts/rd-professional-api/pull/442/commits/a46046fb012a380bc963147eba1c7f3fea65fd22